vue-components/src/components/Lookup.vue

```diff
@@ -123,19 +139,57 @@ export default Vue.extend( {
 		},
 
 		onSelect( menuItem: MenuItem ): void {
-			this.hasItemSelected = true;
+			this.hasActiveElement = true;
 
 			// the following comment generates the event's description for the docs tab in storybook
 			/**
-			 * This even is emitted whenever an item is selected on the Lookup. The event payload contains the whole
+			 * This event is emitted whenever an item is selected on the Lookup. The event payload contains the whole
 			 * MenuItem object.
 			 */
 			this.$emit( 'input', menuItem );
 			this.$emit( 'update:searchInput', menuItem.label );
 
-			// We know that there is one input here because it is part of this component
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			this.$el.querySelector( 'input' )!.blur();
+			// If a menu item is selected using the enter key, keep focus on input component
+			if ( !this.enterKeyPressed ) {
+				// We know that there is one input here because it is part of this component
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				this.$el.querySelector( 'input' )!.blur();
+			}
+		},
+
+		onEnter(): void {
+			if ( this.selectedItemIndex !== -1 ) {
+				this.enterKeyPressed = true;
+				this.onSelect( this.menuItems[ this.selectedItemIndex ] );
+			}
+		},
+
+		onArrowUp(): void {
+			this.selectedItemIndex = Math.max( 0, this.selectedItemIndex - 1 );
+		},
+		onArrowDown(): void {
+			this.selectedItemIndex = Math.min( this.menuItems.length - 1, this.selectedItemIndex + 1 );
+		},
+		onTab(): void {
+			if ( this.selectedItemIndex !== -1 ) {
+				this.onSelect( this.menuItems[ this.selectedItemIndex ] );
+			}
+		},
+		onFocus(): void {
+			this.focused = true;
+			this.escKeyPressed = false;
+			this.enterKeyPressed = false;
+
+			if ( this.value !== null && this.menuItems.length > 0 ) {
+				this.selectedItemIndex = this.menuItems.findIndex(
+					( menuItem ) => { return isEqual( menuItem, this.value ); },
+					this,
+				);
+			}
+		},
+		onEsc(): void {
+			this.escKeyPressed = true;
+			this.selectedItemIndex = -1;
```

Comment:
For some reason (unsure why yet), the selected item is not reset on <kbd>Esc</kbd>. See screencast: https://drive.google.com/file/d/1zly4scjSPBPIfGK_XhOq9Uw0ZAiEs31_/view?usp=sharing

Reply:
Found the reason. The problem is not here, the current code works correctly. The problem is that deselecting a previously selected item by typing again doesn't emit an empty input event to let the consumer know that the item was deselected. Will fix.

Reply:
Fixed. Should work as expected now.

Reply:
Perfect, thank you!